pmd-core/src/main/java/net/sourceforge/pmd/ant/internal/PMDTaskImpl.java

```diff
@@ -240,12 +239,13 @@ private void handleError(RuleContext ctx, Report errorReport, RuntimeException p
         Throwable cause = pmde.getCause();
 
         if (cause != null) {
-            StringWriter strWriter = new StringWriter();
-            PrintWriter printWriter = new PrintWriter(strWriter);
-            cause.printStackTrace(printWriter);
-            project.log(strWriter.toString(), Project.MSG_VERBOSE);
-            IOUtils.closeQuietly(printWriter);
-
+            try (StringWriter strWriter = new StringWriter();
+                 PrintWriter printWriter = new PrintWriter(strWriter)) {
+                cause.printStackTrace(printWriter);
+                project.log(strWriter.toString(), Project.MSG_VERBOSE);
+            } catch (IOException e) {
```

Comment:
Is this catch really needed here? I think it was not there before...
The problem now is, that we would hide maybe a severe error, but would continue anyway... (Ok, not really in this exact case when we manually convert a stacktrace to a string...)

Reply:
Ok, I understand now: While closing, theoretically a IOException could occur. So we need to deal with it.

Then don't use printStackTrace - use `project.log("Error while closing stream: " + e, Project.MSG_ERR);` instead.

Reply:
Done. Should have noticed that this was how the other exceptions were being handled.